src/Microsoft.Azure.Batch.SoftwareEntitlement/TokenVerifier.cs

```diff
@@ -75,14 +75,17 @@ public TokenVerifier WithCurrentInstant(DateTimeOffset instant)
         /// <param name="expectedIssuer">The issuer who should have created the token.</param>
         /// <param name="application">The specific application id of the application </param>
         /// <param name="ipAddress">Address of the machine requesting token validation.</param>
+        /// <param name="cpuCoreCount">The number of CPU cores reported to be found by the application,
+        /// or null if that doesn't need to be verified.</param>
         /// <returns>Either a software entitlement describing the approved entitlement, or errors
         /// explaining why it wasn't approved.</returns>
         public Errorable<NodeEntitlements> Verify(
             string tokenString,
             string expectedAudience,
             string expectedIssuer,
             string application,
-            IPAddress ipAddress)
+            IPAddress ipAddress,
+            int? cpuCoreCount)
```

Comment:
What if we introduced a `ClientContext` type, that lists information from the client like `IpAddress`, `CoreCount` and so on (including `BatchAccount`, `PoolId` etc)?